src/test/testing_assert.cpp

```diff
@@ -103,83 +172,89 @@ bool check_table_equal(const std::shared_ptr<const Table>& opossum_table,
-      const auto expected_column_data_type = data_type_to_string.left.at(expected_table->column_type(col_id));
+    if (opossum_table->column_name(col_id) != expected_table->column_name(col_id)) {
+      const std::string error_type = "Column name mismatch (column " + std::to_string(col_id) + ")";
+      const std::string error_msg = "Actual column name: " + opossum_table->column_name(col_id) + "\n" +
+                                    "Expected column name: " + expected_table->column_name(col_id);
 
-      std::cout << generate_table_comparison() << "Table schema is different."
-                << "Column with ID " << col_id << " is different" << std::endl
-                << "Got: " << opossum_table->column_name(col_id) << " (" << opossum_column_data_type << ")" << std::endl
-                << "Expected: " << expected_table->column_name(col_id) << " (" << expected_column_data_type << ")"
-                << std::endl;
+      std::cout << generate_table_comparison(error_type, error_msg, {{0, col_id}}) << std::endl;
+      return false;
+    }
+
+    if (left_col_type != right_col_type) {
+      const std::string error_type = "Column type mismatch (column " + std::to_string(col_id) + ")";
+      const std::string error_msg =
+          "Actual column type: " + data_type_to_string.left.at(opossum_table->column_type(col_id)) + "\n" +
+          "Expected column type: " + data_type_to_string.left.at(expected_table->column_type(col_id));
+
+      std::cout << generate_table_comparison(error_type, error_msg, {{1, col_id}}) << std::endl;
       return false;
     }
   }
 
   // compare content of tables
   //  - row count for fast failure
   if (opossum_table->row_count() != expected_table->row_count()) {
-    std::cout << "Number of rows is different." << std::endl
-              << "Got: " << opossum_table->row_count() << " rows" << std::endl
-              << "Expected: " << expected_table->row_count() << " rows" << std::endl
-              << generate_table_comparison() << std::endl;
+    const std::string error_type = "Row count mismatch";
+    const std::string error_msg = "Actual number of rows: " + std::to_string(opossum_table->row_count()) + "\n" +
+                                  "Expected number of rows: " + std::to_string(expected_table->row_count());
+
+    std::cout << generate_table_comparison(error_type, error_msg) << std::endl;
     return false;
   }
 
   // sort if order does not matter
   if (order_sensitivity == OrderSensitivity::No) {
-    std::sort(opossum_matrix.begin(), opossum_matrix.end());
-    std::sort(expected_matrix.begin(), expected_matrix.end());
+    // skip header when sorting
+    std::sort(opossum_matrix.begin() + 2, opossum_matrix.end());
+    std::sort(expected_matrix.begin() + 2, expected_matrix.end());
   }
 
-  for (unsigned row = 0; row < opossum_matrix.size(); row++)
+  bool has_error = false;
+  std::vector<std::pair<uint64_t, uint16_t>> mismatched_cells{};
```

Comment:
What do `uint64_t` and `uint16_t` represent? Why are they different?

Reply:
They represent row ids and column ids respectively, and these are their limits, that's why they're different.